arviz/data/io_pyjags.py

```diff
@@ -0,0 +1,323 @@
+
+    """
+    variable_name_to_samples_map = {}
+
+    for key, value in idata.posterior.to_dict()["data_vars"].items():
+        variable_name_to_samples_map[key] = np.array(value["data"])
+
+    return variable_name_to_samples_map
+
+
+def _convert_arviz_sdict_to_pyjags_dict(
+    samples: tp.Mapping[str, np.ndarray]
+) -> tp.Mapping[str, np.ndarray]:
+    """
+    Convert and ArviZ dictionary to a PyJAGS dictionary.
+
+    Takes a python dictionary of samples in ArviZ format and returns the samples
+    as a dictionary in PyJAGS format.
+
+    Parameters
+    ----------
+    samples: a dictionary mapping variable names to NumPy arrays with shape
+             (number_of_chains, chain_length, parameter_dimension)
+
+    Returns
+    -------
+    a dictionary mapping variable names to NumPy arrays with shape
+             (parameter_dimension, chain_length, number_of_chains)
+
+    """
+    # pyjags returns a dictionary of NumPy arrays with shape
+    #         (parameter_dimension, chain_length, number_of_chains)
+    # but arviz expects samples with shape
+    #         (number_of_chains, chain_length, parameter_dimension)
+
+    variable_name_to_samples_map = {}
+
+    for variable_name, chains in samples.items():
+        if chains.ndim == 2:
+            number_of_chains, chain_length = chains.shape
+            chains = chains.reshape((number_of_chains, chain_length, 1))
+
+        variable_name_to_samples_map[variable_name] = np.swapaxes(chains, 0, 2)
+
+    return variable_name_to_samples_map
+
+
+def from_pyjags(
+    posterior: tp.Optional[tp.Mapping[str, np.ndarray]] = None,
+    prior: tp.Optional[tp.Mapping[str, np.ndarray]] = None,
+    coords=None,
+    dims=None,
+    save_warmup=None,
+    warmup_iterations: int = 0,
+) -> InferenceData:
+    """
+    Convert PyJAGS posterior samples to an ArviZ inference data object.
+
+    Takes a python dictionary of samples that has been generated by the sample
+    method of a model instance and returns an Arviz inference data object.
+
+    Parameters
+    ----------
+    posterior: dict of {str : array_like}, optional
+               a dictionary mapping variable names to NumPy arrays containing
+               posterior samples with shape
+               (parameter_dimension, chain_length, number_of_chains)
+
+    prior: dict of {str : array_like}, optional
+               a dictionary mapping variable names to NumPy arrays containing
+               prior samples with shape
+               (parameter_dimension, chain_length, number_of_chains)
+
+    coords: dict[str, iterable]
+            A dictionary containing the values that are used as index. The key
+            is the name of the dimension, the values are the index values.
+
+    dims: dict[str, List(str)]
+          A mapping from variables to a list of coordinate names for the variable.
+
+    save_warmup: whether or not to return warmup draws if any
+    warmup_iterations: how many iterations are for warmup
```

Comment:
follow numpydoc style

Reply:
Ok, it is added.

Reply:
> If possible, we should add 8schools example for the pyjags too.

A pyjags implementation of the eight schools example is in the inference data cookbook.

Reply:
I removed the pyjags requirement again as the azure dev-ops system tries to install the package which fails since it requires the unix package jags underneath (which would have to be installed via apt-get install on Debian Linux and derivatives). Here is the error message:
    Package jags was not found in the pkg-config search path.
    Perhaps you should add the directory containing `jags.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'jags' found
    Error while executing pkg-config: Command '['pkg-config', '--libs', '--cflags', 'jags']' returned non-zero exit status 1.

Reply:
can we do 

    conda install -c conda-forge jags

Reply:
Oh wait, I don't know if we use conda or not in devops.

But adding 

    apt-get update
    apt-get install jags

before line 46 in `.azure-pipelines/azure-pipelines-external.yml` should work

Reply:
Ok, I added the installation via apt-get and re-added pyjags as a requirement.